policy/modules/system/systemd.if

```diff
@@ -58,28 +61,689 @@ template(`systemd_role_template',`
+	manage_chr_files_pattern($1_systemd_t, systemd_user_runtime_t, systemd_user_runtime_t)
+
 	selinux_use_status_page($1_systemd_t)
 
 	seutil_read_file_contexts($1_systemd_t)
 	seutil_search_default_contexts($1_systemd_t)
+
+	systemd_list_journal_dirs($3)
+	systemd_read_journal_files($3)
+
+	init_list_unit_dirs($1_systemd_t)
+
+	systemd_manage_conf_home_content($3)
+	systemd_relabel_conf_home_content($3)
+
+	systemd_manage_data_home_content($3)
+	systemd_relabel_data_home_content($3)
+
+	systemd_search_user_runtime_unit_dirs($1_systemd_t)
+	manage_dirs_pattern($1_systemd_t, systemd_user_runtime_unit_t, systemd_user_runtime_unit_t)
+	manage_files_pattern($1_systemd_t, systemd_user_runtime_unit_t, systemd_user_runtime_unit_t)
+	manage_lnk_files_pattern($1_systemd_t, systemd_user_runtime_unit_t, systemd_user_runtime_unit_t)
+
+	systemd_read_user_unit_files($3)
+	systemd_read_user_unit_files($1_systemd_t)
+
+	systemd_search_user_runtime_unit_dirs($1_systemd_t)
+	systemd_list_user_runtime_unit_dirs($3)
+	systemd_read_user_runtime_units($3)
+
+	systemd_status_user_units($3)
+	systemd_start_user_units($3)
+	systemd_stop_user_units($3)
+	systemd_reload_user_units($3)
+	systemd_status_conf_home_units($3)
+	systemd_start_conf_home_units($3)
+	systemd_stop_conf_home_units($3)
+	systemd_reload_conf_home_units($3)
+
+	type_transition systemd_user_session_type systemd_user_runtime_t:dir systemd_user_runtime_unit_t "generator.early";
+	type_transition systemd_user_session_type systemd_user_runtime_t:dir systemd_user_runtime_unit_t "generator.late";
+	type_transition systemd_user_session_type systemd_user_runtime_t:dir systemd_user_runtime_unit_t "transient";
+	type_transition systemd_user_session_type systemd_user_runtime_t:dir systemd_user_runtime_unit_t "user";
+
+	init_dbus_chat($3)
+	systemd_status_user_runtime_units($3)
+	systemd_start_user_runtime_units($3)
+	systemd_stop_user_runtime_units($3)
+	systemd_reload_user_runtime_units($3)
+
+	optional_policy(`
+		xdg_config_filetrans($1_systemd_t, systemd_conf_home_t, dir, "systemd")
+		xdg_data_filetrans($1_systemd_t, systemd_data_home_t, dir, "systemd")
+		xdg_read_config_files($1_systemd_t)
+		xdg_read_data_files($1_systemd_t)
+	')
+')
+
+######################################
+## <summary>
+##   Allow the specified domain to communicate with the specified
+##   dbusd domain. Should only be used for user systemd types.
+## </summary>
+## <param name="domain">
+##   <summary>
+##     Prefix for the user domain.
+##   </summary>
+## </param>
+## <param name="dbusd_domain">
+##   <summary>
+##     dbusd domain to allow interactions with.
+##   </summary>
+## </param>
+#
+interface(`systemd_user_spec_daemon_domain',`
+	gen_require(`
+		type $1_systemd_t;
+	')
+
+	dbus_spec_domtrans($1_systemd_t, $2)
+	allow $2 $1_systemd_t:unix_stream_socket rw_socket_perms;
+
+	allow $1_systemd_t $2:process signal_perms;
+	allow $1_systemd_t $2:unix_stream_socket { connectto server_stream_socket_perms };
```

Comment:
why this? i dont think this is needed

Reply:
I had some denials from this before, but I'm having trouble reproducing them again. Though, I don't really know what these accesses are for, so it's better safe to remove them until that can be investigated further.

Reply:
This is due to a missing call to `dbus_spec_session_bus_client`. `connectto` should be the only access needed.